File: demo/activex/HelloWorld3.cs
using System;
//using org.xmlBlaster.client.activex;

namespace demo
{
   /// <summary>
   ///   Example how to access xmlBlaster using the ActiveX Bridge
   ///   to access the Java client library (Windows only)
   /// </summary>
   /// <remarks>
   ///   First you need to create and register the ActiveX Bridge
   ///   with the 'build' command below.
   ///   See: xmlBlaster\src\java\org\xmlBlaster\client\activex
   ///   See: build -DJRE_HOME=C:\PROGRA~1\Java\j2re1.5.0 -DJVM.target=1.4 activex
   /// </remarks>
   class HelloWorld3
   {
      internal HelloWorld3() {}

      // TODO: Callback events never arrive
      public string update(object ev) 
      {
         Console.WriteLine("SUUUCEESS: Got update from xmlBlaster: " + ev.ToString());
         return "<qos><state id='OK'/></qos>";
      }

      // TODO: Callback events never arrive
      public string XmlScriptAccessSource_updateEventHandler(object ev) 
      {
         Console.WriteLine("SUUUCEESSSSS: Got update from xmlBlaster: " + ev.ToString());
         return "<qos><state id='OK'/></qos>";
      }

      internal void runTest() 
      {
         // See registry: 
         //   regedit -> HKEY_CLASSES_ROOT
         //           -> CLSID
         //	         -> D824B185-AE3C-11D6-ABF5-00B0D07B8581
         //           -> XmlScriptAccess Bean Control
         // org.xmlBlaster.client.activex.XmlScriptAccess

         XmlScriptAccess.XmlScriptAccessClass xmlBlaster;
         xmlBlaster = new XmlScriptAccess.XmlScriptAccessClass();
         string[] argArr = { "-protocol", "SOCKET"/*,
                             "-trace",    "true"*/ };
         xmlBlaster.initArgs(argArr);
         //Set obj = xmlBlaster.createPropertiesInstance();
         //string request = "<xmlBlaster><connect/></xmlBlaster>";

         
         XmlScriptAccess.XmlScriptAccessSource_updateEventHandler myClass1 = new XmlScriptAccess.XmlScriptAccessSource_updateEventHandler(this.update);
         // Get the type referenced by the specified type handle.
         Type myClass1Type = Type.GetTypeFromHandle(Type.GetTypeHandle(myClass1));
         Console.WriteLine("The Names of the Attributes :"+myClass1Type.Attributes);
                  
         //System.Type type = XmlScriptAccess.XmlScriptAccessSource_updateEventHandler();
         //System.Type type = System.Type.GetType("XmlScriptAccessSource_updateEventHandler"); //UpdateListener");
         XmlScriptAccess.XmlScriptAccessSource_updateEventHandler.CreateDelegate(
            myClass1Type, this, "update");
   
         {
            string request =
               "<xmlBlaster>" +
               "   <connect/>" +
               "   <wait delay='1000' />" +
               "   <publish>" +
               "      <key oid='test'><airport name='london' /></key>" +
               "      <content>This is a simple script test</content>" +
               "      <qos/>" +
               "   </publish>" +
               "</xmlBlaster>";
            string response = xmlBlaster.sendRequest(request);
            Console.WriteLine("Got response from xmlBlaster: " + response);
         }
         {
            string request =
               "<xmlBlaster>" +
               "   <subscribe><key oid='another.topix'></key><qos/></subscribe>" +
               "   <publish>" +
               "      <key oid='another.topix'/>" +
               "      <content>This is a simple script test</content>" +
               "      <qos/>" +
               "   </publish>" +
               "   <wait delay='1000'/>" +
               "</xmlBlaster>";
            string response = xmlBlaster.sendRequest(request);
            Console.WriteLine("Got response from xmlBlaster: " + response);
         }
         {
            string request =
               "<xmlBlaster>" +
               "   <disconnect/>" +
               "</xmlBlaster>";
            string response = xmlBlaster.sendRequest(request);
            Console.WriteLine("Got response from xmlBlaster: " + response);
         }
      }
      

      [STAThread]
      static void Main(string[] args) {
         demo.HelloWorld3 hw = new demo.HelloWorld3();
         hw.runTest();
      }
   }
}

File: xmlBlaster/demo/activex/HelloWorld3.vb
' XmlBlaster access with asynchronous callbacks
' TODO: CALLBACKS ARE NOT ARRIVING
' @author Marcel Ruff
Module HelloWorld3
    Private WithEvents xmlBlaster As XmlScriptAccess.XmlScriptAccessClass
    Sub Main()
        Call HelloWorld3()
    End Sub

    ' This method is never called (event from java): What is wrong?
    Private Sub update(ByVal evt As Object)
        Console.WriteLine("UPDATE ARRIVED:" & evt.ToString())
        MsgBox(evt.ToString)
    End Sub

    ' This method is never called (event from java): What is wrong?
    Private Sub HelloWorld3_update(ByVal evt As Object)
        Console.WriteLine("UPDATE ARRIVED:" & evt.ToString())
        MsgBox(evt.ToString)
    End Sub

    Sub HelloWorld3()
        Dim request, response As String

        'late binding
        'Set objJava = CreateObject("XmlScriptAccess.Bean.1")

        xmlBlaster = New XmlScriptAccess.XmlScriptAccessClass

        'xmlBlaster.addUpdateListener(xmlBlaster)
        '.XmlScriptAccessSource.update()

        ' Configure using the SOCKET protocol
        Dim argArr(1) As String
        argArr(0) = "-protocol"
        argArr(1) = "SOCKET"
        xmlBlaster.initArgs(argArr)
        'MsgBox("Connecting to xmlBlaster ...")
        Try
            ' Connect to the server
            response = xmlBlaster.sendRequest("<xmlBlaster><connect/></xmlBlaster>")

            ' Query the free memory
            request = "<xmlBlaster><subscribe><key oid='test.VB'/></subscribe><publish><key oid='test.VB'/><content>This is a simple Visual Basic script test</content><qos/></publish></xmlBlaster>"
            response = xmlBlaster.sendRequest(request)
            Console.WriteLine("Got response:" & response)

            ' Leave the server
            response = xmlBlaster.sendRequest("<xmlBlaster><disconnect/></xmlBlaster>")
        Catch e As Exception
            Console.WriteLine("Exception:" & e.ToString())
        End Try
    End Sub
End Module
